cityofaustin/atd-vz-data
# Hasura migrations

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/13293

This PR removes the previous manual tracking of Hasura metadata, schema, and migrations. It adds the files generated by [hasura init](https://hasura.io/docs/latest/hasura-cli/commands/hasura_init/). You can see the steps of what I did [here](https://github.com/cityofaustin/atd-data-tech/issues/13293#issuecomment-1692609571).

**Forgot to mention that a ton of these files are autogenerated so I don't think a line by line review is needed. I added some comments, and I think that it really helps to look at the files/structure in a code editor.**

## Testing
**URL to test:** 
Local

**Steps to test:**
1. Checkout this branch, start the virtual environment, and run `./vision-zero replicate-db`
2. Run `./vision-zero hasura-console` to open the console and make the changes in the next steps
3. In the console, go to **Data** > **default** > **public** > **atd_txdot_crashes**
4. Then, go to **Modify** > **Add a new column** to create a column with:
   - column name: **new_column**
   - type: **text**
   - no default value
   - nullable
5. Now, go to the **Permissions** tab and you should see that the **editor** and **readonly** have two cells in the permissions grid with the filter icon
6. Click those filters icons and click the checkbox next to **new_column** under **Column select permissions** and save
7. In your code editor, you should see:
    - `/atd-vz-data/atd-vzd/metadata/databases/default/tables/public_atd_txdot_crashes.yaml` now shows a diff with select permissions for both **editor** and **readonly** roles
    - there is a new migration folder in `/atd-vz-data/atd-vzd/migrations/default` with autogenerated up and down migration files for the new column
8. These would be the changes that we would review in PRs and then apply using the Hasura CLI when merging to `master` and eventually releasing to production. I can write up some docs on that before closing this issue out.

---
#### Ship list
